### PR TITLE
[test] Fix bad wait logic for NOOP DaemonSet

### PR DESCRIPTION
**Pull request by sebsoto**
Fixes an issue in the deletion suite, which was causing tests to fail if the NOOP Daemonset was not immediately deployed. The wait loop was immediately returning on error, instead of continuing to check after the time interval passed.

### CODE REVIEW

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by sebsoto** [comment]
/approve

Self approving as multiple people have LGTM'd, and this is holding up the release-4.13 backports

**Comment by jrvaldes** [comment]
/test nutanix-e2e-operator


ipi installation failed.

**Comment by aravindhp** [comment]
@sebsoto seeing the same [failure](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1763/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-upgrade/1699519798485979136). Please take a look.

**Comment by aravindhp** [comment]
@sebsoto will we start seeing that issue more often because of this fix?

**Comment by sebsoto** [comment]
This fix should have no impact on that issue.
I will look into if it is seen again

**Pull request merged**
